### PR TITLE
ROT-2815 publish sqs messages

### PR DESCRIPTION
**Pull request by mshero**
This PR integrates AWS SQS message publishing into the callback-simulator.

Notes: 
- proper configuration object will be added with a separate PR
- integration test using the docker image will be added with a separate PR

### CODE REVIEW

File: src/main/java/com/ninecookies/wiremock/extensions/util/Objects.java [hunk 1]
```diff
@@ -2,11 +2,17 @@
 
 import java.util.regex.Matcher;
 
+import com.github.tomakehurst.wiremock.common.Json;
 import com.github.tomakehurst.wiremock.http.Response;
 import com.jayway.jsonpath.DocumentContext;
 
 public class Objects {
 
+    public static <T> T convert(Object source, Class<T> destinationType) {
```
Comment (vghero) [inline src/main/java/com/ninecookies/wiremock/extensions/util/Objects.java:11]: Taste: I would drop a javadoc saying that the target type has to have the same structure/fields than source and that it basically copies attributes. Does it fail if attributes are missing?

File: src/test/java/com/ninecookies/wiremock/extensions/util/PlaceholdersKeywordTest.java [hunk 2]
```diff
@@ -59,7 +59,25 @@ public void testTransformUrlWithoutPlaceholders() {
...
+        assertNotNull(value);
+
+        UUID uuid = UUID.fromString(value.toString());
+        assertNotNull(uuid);
+    }
+
+    @Test
+    public void testUUIDPatternInString() {
+
+        System.out.println("pattern: [" + Placeholders.KEYWORD_PATTERN.pattern() + "]");
```
Comment (vghero) [inline src/test/java/com/ninecookies/wiremock/extensions/util/PlaceholdersKeywordTest.java:77]: Leftover :D?

**Review by vghero — APPROVED** [review]
Nice job :+1:

File: src/main/java/com/ninecookies/wiremock/extensions/HttpCallbackHandler.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        }
+    }
+
+    private String readEntity(HttpEntity entity) {
+        String result = null;
+        if (entity != null) {
+            try {
+                result = EntityUtils.toString(entity);
+            } catch (ParseException | IOException e) {
+                /* ignored */
```
[the remark this answers is not in the document]
Comment (vghero) [reply src/main/java/com/ninecookies/wiremock/extensions/HttpCallbackHandler.java:170]: True :+1:

File: src/main/java/com/ninecookies/wiremock/extensions/HttpCallbackHandler.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        authCache.put(host, new BasicScheme());
+        HttpClientContext context = HttpClientContext.create();
+        context.setCredentialsProvider(credentialsProvider);
+        context.setAuthCache(authCache);
+        return context;
+    }
+
+    public static Runnable of(ScheduledExecutorService executor, int maxRetries, int retryBackoff,
+            File callbackFile) {
+        HttpCallbackHandler result = new HttpCallbackHandler();
```
[the remark this answers is not in the document]
Comment (vghero) [reply src/main/java/com/ninecookies/wiremock/extensions/HttpCallbackHandler.java:202]: I would find it strange if all methods in our codebase would have returned objects named 'result` TBH :). But no strong opinion in wiremock :P.

File: src/main/java/com/ninecookies/wiremock/extensions/SqsCallbackHandler.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    public static Runnable of(File callbackFile) {
+        SqsCallbackHandler result = new SqsCallbackHandler();
+        result.callbackFile = callbackFile;
+        return result;
+    }
+
+    @Override
+    public void run() {
+        try (MessagePublisher publisher = MessagePublisher.standard().build()) {
+            SqsCallback callback = readCallback();
```
[the remark this answers is not in the document]
Comment (vghero) [reply src/main/java/com/ninecookies/wiremock/extensions/SqsCallbackHandler.java:47]: ic, makes sense :+1:

File: src/main/java/com/ninecookies/wiremock/extensions/api/Callback.java [hunk 6]
```diff
@@ -21,22 +21,39 @@
...
 
+    /**
+     * Create a new instance for an SQS message {@link Callback} definition.
+     *
+     * @param delay the period of time in milliseconds to wait before the callback data is POSTed.
+     * @param queue the destination queue name to send the message to.
+     * @param message the message to send.
+     * @return a new {@link Callback} instance ready to use.
+     */
+    public static Callback ofQueueMessage(int delay, String queue, Object message) {
```
[the remark this answers is not in the document]
Comment (vghero) [reply src/main/java/com/ninecookies/wiremock/extensions/api/Callback.java:49]: So the answer seems no I guess xD. I would have maybe mentioned it in the @param.... but anyway..

[the remark this answers is not in the document]
Comment (mshero) [reply src/main/java/com/ninecookies/wiremock/extensions/api/Callback.java:49]: In fact it depends on how the JSON library of the client handles an `InputString`. fasterxml Jackson throws an exception `InvalidDefinitionException` for a `ByteArrayInputStream`... xD
Anyway very nit picking. I hear you and I update the docs... ;)

File: src/main/java/com/ninecookies/wiremock/extensions/HttpCallbackHandler.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (mshero) [reply src/main/java/com/ninecookies/wiremock/extensions/HttpCallbackHandler.java:202]: I hear you. I tend to use it for convenient or static construction methods. You suggest `callbackHandler` as the result is an instance of callback handler. Someone else might argue but the result it s `Runnable` it should be named `runnable`... 
For me the implementing or returned type of such a method isn't too important inside of the method and it is quite clear what happens with that very variable.. It's the result regardless of the name...
But that's opinionated...

**Review by devendra-dh — COMMENTED** [review]
nice and explanatory docs ❤️

**Pull request merged**